portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,37 +33,65 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-    private List<String> quotes;
-    private ArrayList<String> messages;
-    @Override
-    public void init() {
-        quotes = new ArrayList<>();
-        quotes.add("The elevator to success is out of order. You'll have to use the stairs, one step at a time.");
-        quotes.add("I always wanted to be somebody, but now I realise I should have been more specific.");
-        quotes.add("I am so clever that sometimes I don't understand a single word of what I am saying");
-        quotes.add("People say nothing is impossible, but I do nothing every day");
-        quotes.add("You can't have everything. Where would you put it?");
-        quotes.add("Well-behaved women seldom make history.");
-    }
-    public ArrayList create_messages() {
-        messages = new ArrayList<>();
-        messages.add("I am a message.");
-        messages.add("Wow that last message is so self aware.");
-        messages.add("Wait, you all can talk?");
-        return messages;
-    }
-    public String toJson(ArrayList<String> alist) {
-        Gson gson = new Gson();
-        String json = gson.toJson(alist);
-        return json;
-    }
-
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String quote = quotes.get((int) (Math.random() * quotes.size()));
-    ArrayList<String> msg_array = create_messages();
-    String jsonMsg = toJson(msg_array);
-    response.setContentType("application/json;");
-    response.getWriter().println(jsonMsg);
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      
+      //HTML Form Steps
+      String first = getParameter(request, "fname", "");
+      String last = getParameter(request, "lname", "");
+      String firstLast = String.format("%s %s",first,last);
+      String comment = getParameter(request, "comment", "");
+
+      HashMap<String, String> name_comment = new HashMap<String, String>();
+      name_comment.put(firstLast, comment);
+
+      //Datastore Steps
+      Entity CommTaskEntity = new Entity("CommentTask");
+      CommTaskEntity.setProperty("First", first);
+      CommTaskEntity.setProperty("Last", last);
+      CommTaskEntity.setProperty("Comment", comment);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(CommTaskEntity);
+
+      response.setContentType("text/html");
+      response.getWriter().println(name_comment);
+
+      response.sendRedirect("/index.html");
   }
+
+// BLOCKED: (explanation for commented lines 23 & 66 - 85)
+// Getting Compling error: "/DataServlet.java:[23,27] package com.google.sps.data does not exist"
```

Comment:
The problem is that the `Task` class doesn't exist in your project. It only exists in the example project. I guess you can copy the class if you want the code below to compile.
In my own implementation, I simply used `String` to represent the comment, but you can also create your own `Comment` class to use instead of `Task` and then replace all instances of `Task` with `Comment`.

Reply:
I just copied and modified the class for my needs in Cloud Shell and after testing, it works fine! How should I apply these changes? Should I finish this PR then make a new one?